packages/flutter/lib/src/painting/shader_warm_up.dart

```diff
@@ -9,26 +9,45 @@ import 'package:flutter/foundation.dart';
 
 /// Interface for drawing an image to warm up Skia shader compilations.
 ///
-/// When Skia first sees a certain type of draw operations on GPU, it needs to
-/// compile the corresponding shader. The compilation can be slow (20ms-200ms).
-/// Having that time as a startup latency is often better than having a jank in
-/// the middle of an animation.
+/// When Skia first sees a certain type of draw operation on the GPU, it needs
+/// to compile the corresponding shader. The compilation can be slow (20ms-
+/// 200ms). Having that time as startup latency is often better than having
+/// jank in the middle of an animation.
 ///
 /// Therefore, we use this during the [PaintingBinding.initInstances] call to
 /// move common shader compilations from animation time to startup time. By
-/// default, a [DefaultShaderWarmUp] is used. Create a custom [ShaderWarmUp]
-/// subclass to replace [PaintingBinding.shaderWarmUp] before
-/// [PaintingBinding.initInstances] is called. Usually, that can be done before
-/// calling [runApp].
+/// default, a [DefaultShaderWarmUp] is used. If needed, app developers can
+/// create a custom [ShaderWarmUp] subclass and hand it to
+/// [PaintingBinding.shaderWarmUp] (so it replaces [DefaultShaderWarmUp])
+/// before [PaintingBinding.initInstances] is called. Usually, that can be
+/// done before calling [runApp].
 ///
-/// This warm up needs to be run on each individual device because the shader
+/// To determine whether a draw operation is useful for warming up shaders,
+/// check the difference in the `worst_frame_rasterizer_time_millis` benchmarks.
```

Comment:
this isn't really part of our public API, i'm not sure people will know what to do with that.
I would say something about looking at the slowest GPU frame in a trace, or something like that.

Reply:
See https://github.com/flutter/flutter/pull/28679